example/src/main/java/com/vimeo/example/MainActivity.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2020 Vimeo (https://vimeo.com)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.vimeo.example
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.vimeo.networking2.*
+import com.vimeo.networking2.config.VimeoApiConfiguration
+import kotlinx.android.synthetic.main.activity_main.*
+
+/**
+ * The sample activity in which we create an [Authenticator] instance to authenticate with the server.
+ */
+class MainActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        // Construct a configuration instance.
+        val configuration = VimeoApiConfiguration.Builder(CLIENT_ID, CLIENT_SECRET, listOf(ScopeType.PUBLIC))
+            .withCodeGrantRedirectUrl("https://example.com")
+            .build()
+
+        // Initialize the authenticator instance with the configuration and the api client.
+        val authenticator = Authenticator(configuration)
+        val apiClient = VimeoApiClient(configuration, authenticator)
+
+        // If we are handling a code grant redirect, this URI will be present, otherwise it's a normal app start.
+        val redirectUri = intent?.data
+
+        if (redirectUri != null) {
+            // Log in using the code grant URI provided to us by the redirect.
+            authenticator.authenticateWithCodeGrant(redirectUri.toString(), vimeoCallback(
```

Comment:
The app fires off an intent to open the code grant URL in the browser. The website prompts the user to log in and authorize the app to access Vimeo. Once that is done, the browser navigates to the redirect URI specified in the configuration. The application has to register that URI in the manifest (see `AndroidManifest.xml` where `https://example.com` is registered in an intent filter). When the browser redirects to `https://example.com?code={auth code}` the app is reopened and the redirect URI is present in the `intent`. The application then extract the URI from the `intent` and pass it to `authenticateWithCodeGrant`, which extracts the authorization code and authenticates with the API. Since `MainActivity` is restarted when a new intent is fired, `onCreate` is called. If `MainActivity` was brought to the front instead, this code would need to be in `onNewIntent`.

The 3 things the user has to do is
1. Set the client ID in this file.
2. Set the client secret in this file.
3. Add `https://example.com` as a callback URL in the API dashboard.

Once those three things are done, this code works as expected.

Reply:
Ah. I see. Thanks for the explanation. I thought I might be missing something. That's definitely something that needs to be called out clearly in the docs. Did you test the entire flow to confirm that it works?

Reply:
Yes this I create my own API app that I used to verify this example while building it. This is the same flow supported by the V1 library.